tests/test_database_sqlite.py

```diff
@@ -1,115 +1,116 @@
-            client = type(database.client).__name__
-            await database.put("hello", {})
-            data = await database.get("hello")
-            await database.delete("hello")
-            await database.disconnect()
-        except NotImplementedError:
-            raise Exception
-        else:
-            self.assertEqual("opsdroid", table)
-            self.assertEqual({}, data)
-            self.assertEqual("Connection", client)
-
-    async def test_deprecated_path(self):
-        database = DatabaseSqlite({"file": "sqlite.db"})
-        assert database.db_file == "sqlite.db"
+
+@pytest.mark.asyncio
+async def test_connect():
+    """Test database connection.
+
+    This method will test the database connection of sqlite database.
+    As the database is created `opsdroid` table is created first.
+
+    """
+    database = DatabaseSqlite({"path": "sqlite.db"})
+    opsdroid = amock.CoroutineMock()
+    opsdroid.eventloop = asyncio.new_event_loop()
+
+    try:
+        await database.connect()
+        table = database.table
+        client = type(database.client).__name__
+        await database.disconnect()
+    except NotImplementedError:
+        raise Exception
+    else:
+        assert table == "opsdroid"
+        assert client == "Connection"
+
+
+@pytest.mark.asyncio
+async def test_disconnect():
+    """Test of database disconnection.
+
+    This method will test the database disconnection of sqlite database.
+
+    """
+    database = DatabaseSqlite({"path": "sqlite.db"})
+    opsdroid = amock.CoroutineMock()
+    opsdroid.eventloop = asyncio.new_event_loop()
+
+    try:
+        await database.connect()
+        await database.disconnect()
+    except NotImplementedError:
+        raise Exception
+    else:
+        pass
+
+
+@pytest.mark.asyncio
+async def test_get_put_and_delete():
+    """Test get, put and delete functions of database
+
+    This method will test the get, put and delete functions which help to read
+    and write data from the database. The function `put` a value with
+    key and asserts the same value after the `get` operation is completed
+    followed by the `delete` operation which deletes the key.
+
+    """
+    database = DatabaseSqlite({"path": "sqlite.db"})
+    opsdroid = amock.CoroutineMock()
+    opsdroid.eventloop = asyncio.new_event_loop()
+
+    try:
+        await database.connect()
+        table = database.table
+        client = type(database.client).__name__
+        await database.put("hello", {})
+        data = await database.get("hello")
+        await database.delete("hello")
+        await database.disconnect()
+    except NotImplementedError:
+        raise Exception
+    else:
+        assert table == "opsdroid"
+        assert data == {}
+        assert client == "Connection"
+
+
+@pytest.mark.asyncio
+async def test_deprecated_path():
+    database = DatabaseSqlite({"file": "sqlite.db"})
+    assert database.db_file == "sqlite.db"
```

Comment:
I know that this wasn't added to the previous tests, but I think we could add another assertion here to check if the logging was shown or not - this is mostly so if we move things around then this test will fail and we know that we need to fix something.

This test would have to look like this:

```python
@pytest.mark.asyncio
async def test_deprecated_path(caplog):
    database = DatabaseSqlite({"file": "sqlite.db"})
    assert database.db_file == "sqlite.db"
    assert "The option 'file' is deprecated, please use 'path' instead." in caplog.text
```